### PR TITLE
Remove devise and oauth2 config

### PR DESCRIPTION
**Pull request by ssendev**

### CODE REVIEW

File: addon/authenticators/devise.js [hunk 1]
```diff
@@ -96,9 +72,11 @@ export default Base.extend({
     @public
   */
   restore(properties) {
-    let propertiesObject = Ember.Object.create(properties);
+    let propertiesObject            = Ember.Object.create(properties);
+    let tokenAttributeName          = this.get('tokenAttributeName');
+    let identificationAttributeName = this.get('identificationAttributeName');
```
Comment (marcoow) [inline addon/authenticators/devise.js]:
```js
const { tokenAttributeName, identificationAttributeName } = this.getProperties('tokenAttributeName', 'identificationAttributeName');
```

File: addon/authenticators/devise.js [hunk 2]
```diff
@@ -122,11 +100,13 @@ export default Base.extend({
   */
   authenticate(credentials) {
     return new Ember.RSVP.Promise((resolve, reject) => {
-      let data                = {};
-      data[this.resourceName] = {
+      let data                        = {};
+      let resourceName                = this.get('resourceName');
+      let identificationAttributeName = this.get('identificationAttributeName');
```
Comment (marcoow) [inline addon/authenticators/devise.js]: see above - use `const { … } = this.getProperties(…)`

File: addon/authenticators/devise.js [hunk 3]
```diff
@@ -156,8 +136,9 @@ export default Base.extend({
     @private
   */
   makeRequest(data) {
+    let serverTokenEndpoint = this.get('serverTokenEndpoint');
```
Comment (marcoow) [inline addon/authenticators/devise.js]: prefer `const`

File: addon/authorizers/base.js [hunk 4]
```diff
@@ -4,9 +4,7 @@ import Ember from 'ember';
   The base for all authorizers. __This serves as a starting point for
   implementing custom authorizers and must not be used directly.__
 
-  __The authorizer preprocesses all XHR requests__ (except ones to 3rd party
-  origins, see
-  [Configuration.crossOriginWhitelist](#SimpleAuth-Configuration-crossOriginWhitelist))
+  __The authorizer preprocesses all XHR requests__
```
Comment (marcoow) [inline addon/authorizers/base.js]: This isn't actually correct anymore neither but a big part of the docs needs to be updated anyway.

**Comment by marcoow** [comment]
Looks good except some style issues (should generally prefer `const` over `let` and use destructuring assignments where possible).

**Comment by ssendev** [comment]
there where some uses of let which could probably converted to const too but i left them how they were. (could chage them too)

File: addon/authenticators/oauth2-password-grant.js [hunk 5]
```diff
@@ -119,9 +95,10 @@ export default Base.extend({
   */
   restore(data) {
     return new Ember.RSVP.Promise((resolve, reject) => {
-      let now = (new Date()).getTime();
+      let now                   = (new Date()).getTime();
```
Comment (marcoow) [inline addon/authenticators/oauth2-password-grant.js]: should be `const` as well

File: addon/authenticators/oauth2-password-grant.js [hunk 6]
```diff
@@ -164,12 +141,13 @@ export default Base.extend({
   */
   authenticate(options) {
     return new Ember.RSVP.Promise((resolve, reject) => {
-      let data = { 'grant_type': 'password', username: options.identification, password: options.password };
+      let data                  = { 'grant_type': 'password', username: options.identification, password: options.password };
```
Comment (marcoow) [inline addon/authenticators/oauth2-password-grant.js]: should be `const` as well

File: addon/authenticators/oauth2-password-grant.js [hunk 7]
```diff
@@ -244,9 +223,10 @@ export default Base.extend({
       dataType:     'json',
       contentType:  'application/x-www-form-urlencoded'
     };
+    const clientId = this.get('clientId');
 
-    if (!Ember.isEmpty(this.clientId)) {
-      let base64ClientId = window.btoa(this.clientId.concat(':'));
+    if (!Ember.isEmpty(clientId)) {
+      let base64ClientId = window.btoa(clientId.concat(':'));
```
Comment (marcoow) [inline addon/authenticators/oauth2-password-grant.js]: should be `const` as well

File: addon/authenticators/oauth2-password-grant.js [hunk 8]
```diff
@@ -283,9 +264,10 @@ export default Base.extend({
     @private
   */
   refreshAccessToken(expiresIn, refreshToken) {
-    let data  = { 'grant_type': 'refresh_token', 'refresh_token': refreshToken };
+    let data                  = { 'grant_type': 'refresh_token', 'refresh_token': refreshToken };
```
Comment (marcoow) [inline addon/authenticators/oauth2-password-grant.js]: should be `const` as well

File: addon/authorizers/devise.js [hunk 9]
```diff
@@ -69,11 +53,12 @@ export default Base.extend({
     @public
   */
   authorize(block) {
+    const { tokenAttributeName, identificationAttributeName } = this.getProperties('tokenAttributeName', 'identificationAttributeName');
     let authenticatedData  = this.get('session.authenticated');
-    let userToken          = authenticatedData[this.tokenAttributeName];
-    let userIdentification = authenticatedData[this.identificationAttributeName];
+    let userToken          = authenticatedData[tokenAttributeName];
+    let userIdentification = authenticatedData[identificationAttributeName];
```
Comment (marcoow) [inline addon/authorizers/devise.js]: these should be `const` as well

**Comment by ssendev** [comment]
i changed the remaining lets to const. (even those that i didn't touch previously)

**Comment by marcoow** [comment]
👍

**Pull request merged**